codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/endpoints/EndpointResolverAdapterGenerator.kt

```diff
@@ -198,8 +213,38 @@ class EndpointResolverAdapterGenerator(
                 continue
             }
 
-            inputContextParams[paramName]?.let {
-                writer.write("builder.#L = input.#L", paramDefaultName, it.defaultName())
+            // Look at input params
+            val inputParam = inputContextParams[paramName]
+            if (inputParam != null) {
+                writer.write("builder.#L = input.#L", paramDefaultName, inputParam.defaultName())
+                continue
+            }
+
+            // Look at operation params
+            val operationParam = operationContextParams?.get(paramName)
+            if (operationParam != null) {
+                val opInputShape = ctx.model.expectShape(op.inputShape)
+                val inputSymbol = ctx.symbolProvider.toSymbol(opInputShape)
```

Comment:
**Style**: The `renderBindOperationContextParams` has gotten too long/complex. We should refactor it to be smaller and use other helper functions, possibly by splitting out the code paths for static params, input params, and op params.